src/types/formTypes.ts

```diff
@@ -157,13 +157,32 @@ export interface CheckedState {
 export type InstitutionDetailsApiCheckedType = CheckedState &
   InstitutionDetailsApiType;
 
+// The pattern excludes characters such as <, >, &, ", $, %, !, #, ?, §, ;, *, ~, \, |, ^, =, [, ], {, }, and ().
+// and ASCII control characters from 0 to 31
+
+const invalidCharactersControlCharactersPattern =
+  // eslint-disable-next-line no-control-regex
+  /^[^<>&"\\\v$%!#?§;*~/\\|^=\\[\]{}()\u0000-\u001F\u007F]+$/u;
+
 export const basicInfoSchema = z.object({
-  firstName: z.string().trim().min(One, {
-    message: CupZodSchemaErrors.firstNameMin,
-  }),
-  lastName: z.string().trim().min(One, {
-    message: CupZodSchemaErrors.lastNameMin,
-  }),
+  firstName: z
+    .string()
+    .trim()
+    .min(One, {
+      message: CupZodSchemaErrors.firstNameMin,
+    })
+    .regex(invalidCharactersControlCharactersPattern, {
+      message: CupZodSchemaErrors.firstNameRegex,
+    }),
+  lastName: z
+    .string()
+    .trim()
+    .min(One, {
+      message: CupZodSchemaErrors.lastNameMin,
```

Comment:
We also need to add the 255 character limit [from the ticket](https://github.com/cfpb/sbl-frontend/issues/911).